pkg/controllers/gitrepository/gitea.go

```diff
@@ -0,0 +1,147 @@
+package gitrepository
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/pkg/controllers/localbuild"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	giteaAdminUsernameKey = "username"
+	giteaAdminPasswordKey = "password"
+)
+
+type GiteaClientFunc func(url string, options ...gitea.ClientOption) (GiteaClient, error)
+
+type giteaProvider struct {
+	client.Client
+	Scheme      *runtime.Scheme
+	giteaClient GiteaClient
+	config      util.CorePackageTemplateConfig
+}
+
+func (g *giteaProvider) createRepository(ctx context.Context, repo *v1alpha1.GitRepository) (repoInfo, error) {
+	resp, _, err := g.giteaClient.CreateRepo(gitea.CreateRepoOption{
+		Name:        getRepositoryName(*repo),
+		Description: fmt.Sprintf("created by Git Repository controller for %s in %s namespace", repo.Name, repo.Namespace),
+		// we should reconsider this when targeting non-local clusters.
+		Private:       false,
+		DefaultBranch: DefaultBranchName,
+		AutoInit:      true,
+	})
+	if err != nil {
+		return repoInfo{}, fmt.Errorf("failed to create git repository: %w", err)
+	}
+	return repoInfo{
+		name:     resp.Name,
+		fullName: resp.FullName,
+		cloneUrl: resp.CloneURL,
+	}, nil
+}
+
+func (g *giteaProvider) getProviderCredentials(ctx context.Context, repo *v1alpha1.GitRepository) (gitProviderCredentials, error) {
+	var secret v1.Secret
+	err := g.Client.Get(ctx, types.NamespacedName{
```

Comment:
Maybe I am off basis here but I wonder if we would be better off getting and storing credentials for the git server from a local machine? I can imagine someone having requirements where storing this in K8s secrets would not be allowed.

Reply:
Great idea. We should support it in the future. Maybe we can discuss it in a feature request issue.

Reply:
@blakeromano mind filing an issue for it?

Reply:
https://github.com/cnoe-io/idpbuilder/issues/237